tests/staticfiles_tests/test_checks.py

```diff
@@ -132,3 +132,54 @@ def test_nonexistent_directories(self):
             # Nonexistent directories are skipped.
             finder = get_finder("django.contrib.staticfiles.finders.FileSystemFinder")
             self.assertEqual(list(finder.list(None)), [])
+
+
+class StorageCheckTests(SimpleTestCase):
+    DEFAULT_STORAGE_VALUE = {
+        "default": {
+            "BACKEND": "django.core.files.storage.FileSystemStorage",
+        },
+        "staticfiles": {
+            "BACKEND": "django.contrib.staticfiles.storage.StaticFilesStorage",
+        },
+    }
+
+    @staticmethod
+    def _remove_storage_alias(alias):
+        """
+        Helper function to remove an alias from the STORAGES setting.
+
+        We need this method because UserSettingsHolder has some backwards compatible
+        code in __setattr__ to keep the aliases 'default' and 'staticfiles' which
+        prevents us from testing what happens when they are missing.
+        """
+        settings.STORAGES.pop(alias, None)
```

Comment:
Why do we need the extra method? "We need this method…" we don't though 🤔 The comment explains why we need to pop instead of constructing a settings with 'default' and 'staticfiles' missing yeah?

I'd vote inlining these to avoid the extra fluff & cognitive load when reading the tests.

Reply:
> Why do we need the extra method? "We need this method…" we don't though 🤔 The comment explains why we need to pop instead of constructing a settings with 'default' and 'staticfiles' missing yeah?

This is true, and an earlier version was doing what you suggest.

My _intent_ was to to document in a single place why we do a `pop()` on the setting dict, instead of using the more official-looking `override_setting` decorator. The decorator doesn't work here, and it took me a while to figure out.

> I'd vote inlining these to avoid the extra fluff & cognitive load when reading the tests.

If I were to inline the call, I imagine that I would duplicate this explanation. Granted, it's only twice now, but if a 3rd/4th test appears later, this would cause further duplication.

I more than happy to rephrase this explanation, which -as you rightly pointed out- isn't explaining properly, but I think there is still value to leave the separate method since it's using an unusual way of overriding the setting.

Reply:
At the end of the day, I would like to explain why we should use `.pop()` instead of `@override_settings`, and explain it once.

Reply:
Yes that's what I figured. 😁

There's no strict dry policy here, in fact we tend to WET oftentimes: Write Everything Twice...

Let's wait to see what felixx & nessita say.

Reply:
My general advice is to make code smart and tests dumb. The dumber the better, in that a test should ideally be as simple and straightforward as possible. Having to chase function calls up and down in a test file has costs, so while sometimes needed, I try to advice to minimize that.

In this case, I think the helper is not needed and that by just calling `pop`, the intent is clear. But, I do think we should restore the setting to the value it originally had?

Reply:
> But, I do think we should restore the setting to the value it originally had?

I don't think we need to worry about this, as -if my understanding is correct- `override_settings` should already take care of that. Let me know if I'm missing something.